src/libcore/string.cpp

```diff
@@ -52,5 +52,12 @@ std::string trim(const std::string &s, const std::string &whitespace) {
     return s.substr(it1, it2 - it1 + 1);
 }
 
+bool contains(const std::vector<std::string> &keys, const std::string &key) {
+    for (auto& k: keys)
```

Comment:
I love ``auto``, but let's reserve its use for complicated types like ``std::some_complex_stl_type<>::iterator`` and spell out types explicitly in all other cases to aid readability. No big deal in this case, but I thought I'd point it out for the future.